src/fal/cli/selectors.py

```diff
@@ -76,6 +76,25 @@ def _filter_node_ids(unique_ids, fal_dbt, selected_nodes, nodeGraph) -> List[str
             if selector_plan.parents:
                 parents = list(nodeGraph.get_predecessors(id))
                 output.extend(parents)
+            if selector_plan.children_with_parents:
+                ids = _get_children_with_parents(id, nodeGraph)
+                output.extend(ids)
+    return output
+
+
+def _get_children_with_parents(node_id, nodeGraph) -> List[str]:
+    output = []
+    children = list(nodeGraph.get_descendants(node_id))
+    output.extend(children)
+    all_parents = []
+    for child in children:
+        parents = list(nodeGraph.get_ancestors(child))
+        all_parents.extend(parents)
```

Comment:
implement this one with reduce

Reply:
is this:
```
reduce(
  lambda x: list.extend(list(nodeGraph.get_ancestors(x))),
  children,
  all_parents,
)
```
more readable that this:
```
for child in children:
  parents = list(nodeGraph.get_ancestors(child))
  all_parents.extend(parents)
```
?

🤷

Reply:
what about
```py
parentss = map(lambda child: nodeGraph.get_ancestors(child), children)
all_parents = reduce(lambda x, y: x + y, parentss, all_parents)
```

Reply:
still, that for is readable, just practicing